pkg/authorize/jwt/validator.go

```diff
@@ -0,0 +1,77 @@
+package jwt
+
+import (
+	"errors"
+	"log"
+
+	"github.com/openshift/telemeter/pkg/authorize"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+// Validator is called by the JWT token authentictaor to apply domain specific
+// validation to a token and extract user information.
+type Validator interface {
+	// Validate validates a token and returns user information or an error.
+	// Validator can assume that the issuer and signature of a token are already
+	// verified when this function is called.
+	Validate(tokenData string, public *jwt.Claims, private interface{}) (*authorize.Client, error)
```

Comment:
the GoDoc for the `Validator` interface says this called by the token authenticator, but the type returned by this method is a `*authorize.Client`, maybe we should refactor this into a more generic `auth` package, or split authn and authz into two distinct packaged